### PR TITLE
V1.10

### PR DESCRIPTION
**Pull request by matannahmani**
This is a huge PR, that changes how we use our system prompt, tools, and UI.
It's the last big PR we will do before rewriting the entire project into a core project that can be consumed on many different UI's example: Web, APP, Extension and more to come :)

Key changes:

- [x] - New UI for tools
- [x] - Only use ASK for tools and update the ASK (this will be used later to remove the abstraction of say and ask)
- [x] - Add Diff Stream Viewer (imported from Cline, will be deprecated shortly after for a better successor)
- [x] - Updated system prompt that removes "openai" / "anthropic" baked in function calling in favor of XML tags and manually parsing them.
- [x] - Tool Parser (parse XML stream and callback onToolUpdate and onToolEnd, etc...)
- [x] - Tool Executor revamped (refactor to better suit our new format of Tool Parser)
- [x] - Context Window bug fixes
- [x] - Built-in task memory management
- [x] - Built-in version control (automatically track milestones and commit them)
- [x] - Collapsable Task Header
- [ ] - Built-in terminal manager using Execa instead of using vs code shell integration (currently has many bugs and is unstable)
- [ ] - New Automatic mode (reworked from the ground up)
- [x] - Personalized prompting experience (based on user technical level)

### CODE REVIEW

**Comment by matannahmani** [comment]
**notes**
- currently write to file is broken, requires minor fixing
- missing a few changes on system prompt (memory management)
- git handlers are turned off temporarily
- new automatic mode - isn't merged yet

File: extension/src/agent/v1/system-prompt.ts [hunk 1]
```diff
@@ -37,10 +37,12 @@ Guidelines:
 Additional Instructions:
 
 - Web Project Bootstrapping:
-  - Use the boilerplate at https://github.com/kodu-ai/kodu-remix-shadcn to bootstrap web projects. It creates a Remix web app with shadcn and Tailwind CSS.
+  - You must use https://github.com/kodu-ai/kodu-remix-shadcn.git to bootstrap web projects. It creates a Remix V2 project with the following tech stack: remix version 2, vite (includes hotreload), shadcn ui, Tailwind CSS and framer motion it's a solid fundation for any web project!
   - To use it:
     git clone https://github.com/kodu-ai/kodu-remix-shadcn.git <project-name> && cd <project-name> && npm install
     then use list_files to understand the project structure and read the root files to see what's inside.
+    from the momemnt you clone the project you should start writing to your memory and remember at all time that you project is located at <project-name> folder.
```
Comment (chris-miaskowski) [inline extension/src/agent/v1/system-prompt.ts:44]:
```suggestion
    from the moment you clone the project you should start writing to your memory and remember at all time that you project is located at <project-name> folder.
```

Comment (ajnart) [reply extension/src/agent/v1/system-prompt.ts]: Thank you for noticing this @chris-miaskowski !

**Comment by matannahmani** [comment]
merged.

**Pull request closed without merging**